src/components/app-root/MapsTabBody/MapsTabBody.js

```diff
@@ -130,10 +135,29 @@ export default class MapsTabBody extends React.PureComponent {
       isOpen: false,
     },
     resizeCount: 0,
-    baselineMapRef: undefined,
-    projectedMapRef: undefined,
+    baselineMapInstance: null,
+    projectedMapInstance: null,
+  };
+
+  setBaselineMapInstance = (mapInstance) => {
+    if (!this.state.baselineMapInstance) {
+      this.setState({ baselineMapInstance: mapInstance }, () => {
+        if (this.state.bounds) {
+          this.updateViewportFromBounds(this.state.bounds);
+        }
+      });
+    }
   };
 
+  setProjectedMapInstance = (mapInstance) => {
+    if (!this.state.projectedMapInstance) {
+      this.setState({ projectedMapInstance: mapInstance }, () => {
+        if (this.state.bounds) {
+          this.updateViewportFromBounds(this.state.bounds);
+        }
+      });
+    }
+  };
```

Comment:
I know I wrote the original code that this derives from, but old or new, this is quite tricky. (And that is a bit of a code smell to me; there ought to be some way to do this better ... but that's a question for another day.) My point: A bit more commenting/documentation about just what this does, from a higher-level viewpoint, might be helpful.

Reply:
_If you have the time and willingness_, I think some simplification can be achieved by converting this class-based component to a functional component with hooks. In particular, that could simplify or even eliminate some of the gyrations needed in the lifecycle callbacks, which are really complicated.